accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,106 @@
+- Each node presented the Nodes information is a link to the system overview page of that node
+
+## Documentation
+
+In addition to all the above, we are enriching our documentation on the CaaS Platform deployment. In the following sections, we are going to describe what we should introduce in the documentation.
+
+### AutoYaST Deployment documentation
+
+CaaS Platform provides AutoYaST templates for deploying a standard cluster (in the `SUSE-CaaSP-Management` package) comprising of:
+
+- 3 Masters
+- 2 Workers
+
+In the Uyuni/SUSE Manager documentation, we need to provide a step-by-step guide on how to customize the AutoYaST template, import it in Uyuni/SUSE Manager and deploy it to bare-metal machines.
+
+## Next steps
+
+Possible next steps are presented as additional "stretch goals": if time permits, those goals can be achieved in the scope of the initial integration.
+
+### Minion blackout configuration
+
+If the [minion blackout](https://docs.saltstack.com/en/latest/topics/blackout/) is feasible, we can set to blackout every minion registered to Uyuni/SUSE Manager that is part of a Kubernetes or CaaS Platform cluster. Every minion that is in blackout mode will reject all incoming commands.
+In case we want to debug the minion, we might want to either:
+
+- Remove the blackout when we need to debug
+- Whitelist the additional functions that will be allowed during blackout for debugging.
+
+### Container registry documentation
+
+In the documentation, [we can link to the CaaS Platform documentation to explain how to setup a local container registry and mirror the SUSE one](https://github.com/SUSE/doc-caasp/blob/master/adoc/admin-crio-registries.adoc#mirror).
+
+### Management node
+
+As part of the support for the CaaS Platform version 4, SUSE Manager can elect (via the Add-On System Type) a node as "Management Node" for the cluster. This node must run SUSE Linux Enterprise Server and have `skuba` (a tool packaged by CaaS Platform) to deploy and manage the systems comprising the cluster.
+
+`skuba` is not packaged for OpenSUSE at the time of writing: Uyuni support will be implemented but enabled later, as soon as the package is released for OpenSUSE.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+If a user does not read the documentation, a Kubernetes cluster can be broken by simply rebooting a node from Uyuni/SUSE Manager.
+
+# Alternatives
+[alternatives]: #alternatives
+We can choose to run `skuba` on the SUSE Manager server, we should add it to the packaging pattern of SUSE Manager server. The package is Go-vendored, no other dependency is required.
+The problem of this approach is that the CaaS Platform admin needs shell access to the Uyuni/SUSE Manager server in order to access and run `skuba`.
+However, shell access to the Uyuni/SUSE Manager server could also be used for other (malicious) things beyond cluster management. And it breaks separation of concerns.
+For the reasons above, this approach was not followed.
```

Comment:
I see, I would still allow SUMA node to be selected by the user as "CaaSP management node" though.

Reply:
Yes, I think in some cases it would still make sense to have SUMA be also the node from which skuba has to be run. There are circumstances where the operator of SUMA is also the operator of the k8s cluster.

Reply:
IMHO auto-registering Uyuni/SUSE Manager to itself (installing and registering salt-minion on the same machine running the salt-master) might not probably be a good solution.
In these special cases, you are mentioning we can maybe add a note in the documentation saying that to fulfill these special needs users can install `skuba` on the SUSE Manager itself, provided they have shell access. What do you think?

Reply:
Actually, most non-SUSE Manager Salt setups will do exactly that: Make the master a minion, too. I really want to see this going forward. Off topic for this particular RfC. But as long as we are careful with a few things (e.g. not let the Manager Server patch itself from its own repos), that would allow some simplifications that would make a lot of sense. E.g., currently we need to call salt-call on the master to use Salt for setting up the "self-monitoring" exporters.